packages/amplify_core/lib/src/plugin/amplify_plugin_key.dart

```diff
@@ -104,3 +104,39 @@ abstract class AuthPluginKey<
   /// {@macro amplify_core.plugin.auth_plugin_key}
   const AuthPluginKey();
 }
+
+// TODO(HuiSF): Define generic parameters for all API types
```

Comment:
I think as they're migrated makes sense. Things might change during the migration.